core/templates/pages/exploration-player-page/templates/lesson-information-card-modal.component.ts

```diff
@@ -166,4 +179,36 @@ export class LessonInformationCardModalComponent extends ConfirmOrCancelModal {
   isLanguageRTL(): boolean {
     return this.i18nLanguageCodeService.isCurrentLanguageRTL();
   }
+
+  async saveLoggedOutProgress(): Promise<void> {
+    if (!this.loggedOutProgressUniqueUrlId) {
+      this.explorationPlayerStateService
+        .setUniqueProgressUrlId()
+        .then(() => {
+          this.loggedOutProgressUniqueUrlId = (
+            this.explorationPlayerStateService.getUniqueProgressUrlId());
+          this.loggedOutProgressUniqueUrl = (
+            this.urlService.getOrigin() +
+            '/progress/' + this.loggedOutProgressUniqueUrlId);
+        });
+    }
+    this.saveProgressMenuIsShown = true;
```

Comment:
Does this depend on the successful completion of the call above?

Reply:
No, it does not. It is simply there to toggle to save progress menu -- whether it is shown or not.

Reply:
Should the menu be shown / hidden when the call above fails?

Reply:
No. The call is meant to be executed only once to fetch the logged out progress URL ID from the backend. For the rest of the cases, this function would only toggle the save progress menu. Regarding the failure of the above call due to any internal server error, we would still want the user to see the save progress menu which contains a prompt that his progress will be saved automatically if he signs in.